### PR TITLE
Use smoke tests from n-test

### PR DESCRIPTION
**Pull request by adgad**
For backwards compatibility, switch out the smoke tests in n-heroku-tools for n-test.

In an ideal world there wouldn't be a dependency here. I think local smoke tests can start using n-test directly. The deploy checks will be harder to unpick, so will create an issue to mull over.

### CODE REVIEW

File: tasks/deploy.js [hunk 1]
```diff
@@ -87,7 +91,7 @@ module.exports = function (program, utils) {
 		.command('deploy [app]')
 		.description('runs haikro deployment scripts with sensible defaults for Next projects')
 		.option('-s, --skip-gtg', 'skip the good-to-go HTTP check')
-		.option('-a, --authenticated-smoke-tests', 'authenticate smoke tests with a backend authorization key')
+		.option('-a, --authenticated-moke-tests', 'authenticate smoke tests with a backend authorization key')
```
Comment (i-like-robots) [inline tasks/deploy.js]: moke

**Comment by sjparkinson** [comment]
Any chance your changes mean we can look to remove `nightwatch`?

It's the cause of the whitesource security things, it depends on the broken ejs package 😢 .

File: lib/smoke-test.js [hunk 2]
```diff
@@ -1,277 +0,0 @@
-'use strict';
```
Comment (wheresrhys) [inline lib/smoke-test.js]: OMG, I'm getting all emotional seeing this file die :cry:

Comment (adgad) [reply lib/smoke-test.js]: Death is just another path, one that we all must take. The grey rain-curtain of this world rolls back, and all turns to silver glass, and then you see it.

Comment (adgad) [reply lib/smoke-test.js]: (sorry I was watching Lord of the Rings the other day)

Comment (wheresrhys) [reply lib/smoke-test.js]: Would you like to borrow my scythe and cowl?

**Comment by adgad** [comment]
@sjparkinson Not quite yet, because this doesn't do cross browser.

But I did create this - https://github.com/Financial-Times/n-test/issues/10 - would like to have another command that takes the same /test/smoke.js but run against cross browser tests. @rowanbeentje mentioned the apps team use http://webdriver.io/ so could look into that.

**Comment by i-like-robots** [comment]
The name "Webdriver" gives me nightmares.

**Comment by rowanbeentje** [comment]
😀 WebDriver has so many problems, but webdriver.io tries hard to paper them over - it has a nice promise-based API, and will handle a lot of the "stale element reference" type errors for you automatically.

**Pull request merged**